uncomment-locales.pl

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/perl -i
+
+my %uniqLocs = ();
+
+for my $subDir ('icinga-L10n', 'icingaweb2/modules/*/application') {
+	for my $lcDir (glob "/usr/share/$subDir/locale/*_*") {
+		if ($lcDir =~ /(\w+)$/) {
+			$uniqLocs{$1} = 1
+		}
+	}
+}
+
+my @locs = keys %uniqLocs;
+
+while (<>) {
+	if (/\bUTF-8\b/) {
+		for my $loc (@locs) {
+			if (/\b$loc\b/) {
+				s/^# *//;
+				last
+			}
+		}
+	}
+} continue {
+	print or die $!
+}
```

Comment:
Does this do anything significantly different than the following would do?

```sh
basename -a /usr/share/{icinga-L10n,icingaweb2/modules/*/application}/locale/*_* | sort -u | sed 's/$/ UTF-8/' > /etc/locale.gen
```

Reply:
## Your suggestion

```
ar_SA UTF-8
de_DE UTF-8
es_AR UTF-8
fi_FI UTF-8
fr_FR UTF-8
it_IT UTF-8
ja_JP UTF-8
pt_BR UTF-8
ru_RU UTF-8
uk_UA UTF-8
```

## My changeset (comments filtered out)

```
ar_SA.UTF-8 UTF-8
de_DE.UTF-8 UTF-8
es_AR.UTF-8 UTF-8
fi_FI.UTF-8 UTF-8
fr_FR.UTF-8 UTF-8
it_IT.UTF-8 UTF-8
ja_JP.UTF-8 UTF-8
pt_BR.UTF-8 UTF-8
ru_RU.UTF-8 UTF-8
uk_UA.UTF-8 UTF-8
```

## Rationale

* Don't care about the above naming schema, nor keep it in sync, it's /etc/locale.gen's job
* Let other, commented, locales where they are, for derived images
* Don't populate bleeding edge locale already supported by IW2, but not the currently used Debian

Reply:
How does Icinga Web even interact with system locales?

> Don't populate bleeding edge locale already supported by IW2, but not the currently used Debian

Apart from the question how likely this is, what would that mean for that locale in Icinga Web, what would work, what wouldn't?

Reply:
1. Didn't dig deep into that. Well, it does setlocale() based on my browser language. Locale not present – setlocale() fails.
2. The locale in question would just work on newest Fedora/whatever, but just not work in our image until base image update. With this PR. I guess.

Reply:
I'd still prefer using shell over perl here. I think more people interacting with this repo will understand this and also, this will save the extra build stage, i.e. less cluttered the Dockerfile.

My first suggestion was based on how `locale.gen` looks on my machine, obviously it's slightly different on debian, but adapting this shouldn't be an issue.

If you want to keep the file as close to the original as possible, the following should really do exactly the same as your perl script:

```sh
basename -a /usr/share/{icinga-L10n,icingaweb2/modules/*/application}/locale/*_* | while read -r l; do sed -i "/$l.*UTF-8/ s/^# *//" /etc/locale.gen; done
```

Reply:
Cool, this meets all of my requirements and is even smaller by an order of magnitude! Feel free to PR it w/ closes #103.